pyproject.toml

```
@@ -96,16 +96,11 @@ build-backend = "poetry.core.masonry.api"
 [tool.pyright]
 
 [tool.ruff]
```

Actually, as an intermediary step, I'd suggest to use
 ```suggestion
[tool.ruff]
exclude = ["*.pyi"]
```
This way the pyi files can still be generated as it is now, (using black in script mode)
but we can use ruff for everything else, reducing the amount of changes we have to go through.

Once we are up to date with latest ruff we can consider dropping the use of black in `.pyi` files

---

updated and resolved conflicts :)

---

Thanks, seems like the change to the pyi (from earlier commits?) are still present, if we remove them I think we'll be good.

---

revered and fixing some mnors